demo/admin/src/products/future/generated/ProductForm.tsx

```diff
@@ -68,7 +80,7 @@ export function ProductForm({ id }: FormProps): React.ReactElement {
         () =>
             data?.product
                 ? {
-                      ...filterByFragment<GQLProductFormDetailsFragment>(productFormFragment, data.product),
+                      ...filterByFragment<ProductFormDetailsFragment>(productFormFragment, data.product),
```

Comment:
this is not correct I think, the return value for filterByFragment is GQLProductFormDetailsFragment, not the custom ProductFormDetailsFragment

Reply:
This is a workaround because the types are not generated correctly, as `finalFormFileUploadFragment` comes from the `@comet/cms-admin` package. 

We discussed this here: https://github.com/vivid-planet/comet/pull/2151#discussion_r1668076984

I then added the same workaround to the handmade `ProductsForm`. 

https://github.com/vivid-planet/comet/blob/de44f4699e8dbed5097abb3e3f41e4bac980c27b/demo/admin/src/products/ProductForm.tsx#L65-L69

Once we have a solution for the type-generation we can remove the workaround from both, the handmade and generated `ProductForm` components. But I think this is something for a separate PR.

Reply:
:(

Reply:
I'm not sure if we can achieve an elegant solution here without the fragment being in the application code... Maybe generate the fragment instead of exporting it from the library?

Reply:
> Maybe generate the fragment instead of exporting it from the library?

So, should we do this? 
In the handmade example as well then?

Reply:
We can change this in a follow-up PR